src/System.Device.Gpio/System/Device/Gpio/Drivers/LibGpiodDriver.Linux.cs

```diff
@@ -0,0 +1,149 @@
+        {
+            // Libgpiod Api do not support pull up or pull down resistors for now.
+            return mode != PinMode.InputPullDown && mode != PinMode.InputPullUp;
+        }
+
+        protected internal override void OpenPin(int pinNumber)
+        {
+            SafeLineHandle pinHandle = Interop.GetChipLineByOffset(_chip, pinNumber);
+            if (pinHandle == null)
+            {
+                throw new IOException($"Pin number {pinNumber} not available for chip: {_chip} error: {Marshal.GetLastWin32Error()}");
+            }
+            _pinNumberToSafeLineHandle.Add(pinNumber, pinHandle);
+        }
+
+        protected internal override PinValue Read(int pinNumber)
+        {
+            if (_pinNumberToSafeLineHandle.TryGetValue(pinNumber, out SafeLineHandle pinHandle))
+            {
+                int value = Interop.GetGpiodLineValue(pinHandle);
+                if (value == -1)
+                {
+                    throw new IOException($"Error while reading value from pin number: {pinNumber}, error: {Marshal.GetLastWin32Error()}");
+                }
+                return value;
+            }
+            throw new IOException($"Error while reading value from pin number: {pinNumber}");
+        }
+
+        protected internal override void RemoveCallbackForPinValueChangedEvent(int pinNumber, PinChangeEventHandler callback)
+        {
+            throw new NotImplementedException();
+        }
+
+        protected internal override void SetPinMode(int pinNumber, PinMode mode)
+        {
+            int success = -1;
+            if (_pinNumberToSafeLineHandle.TryGetValue(pinNumber, out SafeLineHandle pinHandle))
+            {
+                if (mode == PinMode.Input)
+                {
+                    success = Interop.RequestLineInput(pinHandle, pinNumber.ToString());
+                }
+                else
+                {
+                    success = Interop.RequestLineOutput(pinHandle, pinNumber.ToString());
+                }
+            }
+
+            if (success == -1) {
+                throw new IOException($"Error setting pin mode, pin:{pinNumber}, error: {Marshal.GetLastWin32Error()}");
+            } 
+        }
+
+        protected internal override WaitForEventResult WaitForEvent(int pinNumber, PinEventTypes eventType, CancellationToken cancellationToken)
+        {
+            throw new NotImplementedException();
+        }
+
+        protected internal override void Write(int pinNumber, PinValue value)
+        {
+            if (_pinNumberToSafeLineHandle.TryGetValue(pinNumber, out SafeLineHandle pinHandle))
+            {
+                Interop.SetGpiodLineValue(pinHandle, (value == PinValue.High) ? 1 : 0);
+            }
+        }
+
+        protected override void Dispose(bool disposing)
+        {
+            foreach (SafeLineHandle pinHandle in _pinNumberToSafeLineHandle.Values)
```

Comment:
We should be emptying out the dictionary here. In order to do that, you will probably want to use while loop instead where you pop the next item, and release it and remove it from the collection.